### PR TITLE
Added borders and hovers

### PR DESCRIPTION
**Pull request by joszz**
Ability to add border and hover for the round image. Also integrated
hovers on fallback image

### CODE REVIEW

**Comment by monobasic** [comment]
Hey @joszz! Many thanks for your contribution, mucho appreciated! However I don't want to merge it because of the following reasons:
- Border and hover styles are styles, means they belong to the CSS and can be targeted with the usual CSS selectors
- JS configuration options for a jQuery plugin should be implemented via a passed in configuration object, there is too much overhead logic going on in your commit
- Basically what you do is generating options via data attributes, then the js logic parses the data attributes and generates a embeded css style tag, what you do imo is re-inventing the class/css logic. Instead you could apply just some classes to the image, and set the styles via CSS
- Much better would be a function that fetches all applied classes from the img and attaches it later to the svg tag

**Comment by joszz** [comment]
Thx for the suggestions @monobasic.
Good points! Just refactored a bit so it's closer to your suggestions. To bad there's no solid way of getting the value of :hover psuedo class defined in CSS so I left that in there. The onmouseover/onmouseout need to be in there for gracefull fallback (IE8 for example)

There's also no good way of retrieving the values of this CSS specifically;
border: 1px solid green
In jQuery you can only access border-left etc. So I picked left for all of them (which is set in above example). Just see if you want to do anything with it.

I'm still a bit of a noob when it comes to jQuery plugin design :p

**Comment by monobasic** [comment]
I think I don't understand what you try to achive: You could use svg.my-svg:hover as example in CSS to target the hover state of a svg. The styling or "decoration" should go where it belongs - in the CSS - not in a general js method to replace images with svg markup. The cleanest solution imo would be to have classes from the image element automatically added to the generated svg markup.

**Comment by joszz** [comment]
I agree with your puristic approach, but what you're suggesting is not working, as far as my knowledge goes at least.
It's a round image, what I want is a round border + the ability to have the round border color change on hover. And also having the fallback hover work, for example IE8 which cannot render SVG (I looked into VML support but that seems to be too much hassle for what it's worth).
If I use your CSS selector it will create a square border around the SVG block element. Not what I was intending to do. As far as I know you cannot style SVG circle elements with CSS (for example svg.mysvg circle:hover)

But anyways, just see what you do with it. I need this support for a project, and even though it's not the shiniest of solutions it works, and works well crossbrowser/platform.

**Comment by monobasic** [comment]
Ok I see now. As for the SVG solution it would be possible to add another circle just for the border: Check this out: https://jsfiddle.net/monobasic/Lwg9rkku/1/ - however you would still need IE fallback. By the way, depending on your needed targeted IE versions, you could just use a img tag with css border-radius 50% to round it.
